### PR TITLE
Enhance `Cursor.description` with type information

### PR DESCRIPTION
**Pull request by mdesmet**
## Description

Add internal_size, precision, scale to `Cursor.description` to make it easier to get type information of a result set.

Note that if this information can be extracted from the type, this is also removed from the type_code, making it easier to match on.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
* Enhance `Cursor.description` with type information
```

### CODE REVIEW

File: trino/constants.py [hunk 1]
```diff
@@ -50,3 +50,7 @@
 
 HEADER_SET_SCHEMA = "X-Trino-Set-Schema"
 HEADER_SET_CATALOG = "X-Trino-Set-Catalog"
+
+LENGTH_TYPES = ["varchar"]
```
Comment (hovaesco) [inline trino/constants.py:54]: `char` is missing.

File: trino/dbapi.py [hunk 2]
```diff
@@ -220,6 +221,31 @@ def cursor(self, experimental_python_types: bool = None):
...
+        type_signature = column["typeSignature"]
+        raw_type = type_signature["rawType"]
+        arguments = type_signature["arguments"]
+        return cls(
+            column["name"],  # name
+            raw_type if raw_type in PRECISION_TYPES + LENGTH_TYPES else column["type"],  # type_code
+            None,  # display_size
+            arguments[0]["value"] if len(arguments) >= 1 and raw_type in LENGTH_TYPES else None,  # internal_size
+            arguments[0]["value"] if len(arguments) >= 1 and raw_type in PRECISION_TYPES else None,  # precision
+            arguments[1]["value"] if len(arguments) >= 2 and raw_type in SCALE_TYPES else None,  # scale
```
Comment (hovaesco) [inline trino/dbapi.py:244]: why `len(arguments) >= 2` ?

Comment (mdesmet) [reply trino/dbapi.py:244]: We can drop this all together actually as we check on the types itself already.

**Comment by mdesmet** [comment]
This can be looked at again after `PARAMETRIC_DATETIME` is merged as now no type information is exposed for temporal types.

**Pull request closed without merging**